File: 0001-two-sum/solution.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {

        unordered_map<int, int> seen;

        for (int index = 0; index < nums.size(); index++) {
            int value = nums[index];
            int complement = target - value;
            auto it = seen.find(complement);
            if (it != seen.end()) {
                return { it->second, index };
            }
            seen[value] = index;
        }
        return {};
    }
};

File: 0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        self.seen = {}
        for index, value in enumerate(nums):
            complement = target - value
            if complement in self.seen:
                return [self.seen[complement], index]
            self.seen[value] = index
        return []
